pkg/util/validation/limits_test.go

```diff
@@ -110,6 +112,42 @@ func TestLimitsLoadingFromYaml(t *testing.T) {
 	assert.Equal(t, 100, l.MaxLabelNameLength, "from defaults")
 }
 
+func TestLimitsLoadingFromJson(t *testing.T) {
+	SetDefaultLimitsForYAMLUnmarshalling(Limits{
+		MaxLabelNameLength: 100,
+	})
+
+	inp := `{"ingestion_rate": 0.5}`
+
+	l := Limits{}
+	err := json.Unmarshal([]byte(inp), &l)
+	require.NoError(t, err)
+
+	assert.Equal(t, 0.5, l.IngestionRate, "from json")
+	assert.Equal(t, 100, l.MaxLabelNameLength, "from defaults")
+}
+
+func TestLimitsTagsYamlMatchJson(t *testing.T) {
+	limits := reflect.TypeOf(Limits{})
+	n := limits.NumField()
+	var mismatch []string
+
+	for i := 0; i < n; i++ {
+		field := limits.Field(i)
+
+		// Note that we aren't requiring YAML and JSON tags to match, just that
+		// they either both exist or both don't exist.
+		hasYamlTag := field.Tag.Get("yaml") != ""
+		hasJsonTag := field.Tag.Get("json") != ""
```

Comment:
We have a picky linter :)

```suggestion
		hasJSONTag := field.Tag.Get("json") != ""
```